src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -139,33 +144,83 @@ func (col *Collection) populateByOptionIdentifier(
 		return
 	}
 
+	limitCh := make(chan struct{}, urlPrefetchChannelBufferSize)
+	defer close(limitCh)
+
+	type uerr struct {
+		user string
+		err  error
+	}
+
+	errCh := make(chan uerr, len(col.jobs))
+	defer close(errCh)
+
+	ffail := int64(0)
 	for _, identifier := range col.jobs {
-		response, err := query(ctx, col.service, user, identifier)
-		if err != nil {
-			errs = support.WrapAndAppendf(user, err, errs)
+		if ffail > 0 {
+			break
+		}
+		limitCh <- struct{}{}
+
+		wg.Add(1)
+
+		go func(identifier string) {
+			defer func() { <-limitCh }()
+			defer wg.Done()
+
+			var (
+				response absser.Parsable
+				err      error
+			)
+
+			for i := 0; i < numberOfRetries; i++ {
+				response, err = query(ctx, col.service, user, identifier)
+				if err == nil {
+					break
+				}
+				// TODO: Tweak sleep times
+				if i != numberOfRetries-1 {
```

Comment:
nit: bit of a weird comparison.  I'd rather the code make more sense, even if the const seems off by one.
```suggestion
				if i >= numberOfRetries {
```

Reply:
I don't think `i >= numberOfRetries` will work. I think I'll probably change the code to the following to make it more explicit though:

```go
for i := 1; i <= numberOfRetries; i++ {
    ...
    if i < numberOfRetries {
        time.Sleep(time.Duration(3*(i+1)) * time.Second)
    }
}
```

Reply:
Hah, good catch.  My brain was backwards.

Reply:
The reason for the conditional was so that it would not sleep even with an error in the last run. If we skip that, it sleeps at the end of `n-1th` run before exiting out of the loop.